sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/FeatureManagementProperties.java

```diff
@@ -105,13 +92,57 @@ private void addToFeatures(Map<? extends String, ? extends Object> features, Str
         }
     }
 
+    private void addServerSideFeature(Map<? extends String, ? extends Object> features, String key, String combined) {
+        final Object featureValue = features.get(key);
+        if (!combined.isEmpty() && !combined.endsWith(".")) {
+            combined += ".";
+        }
+
+        ServerSideFeature serverSideFeature = null;
+        try {
+            serverSideFeature = MAPPER.convertValue(featureValue, ServerSideFeature.class);
+        } catch (IllegalArgumentException e) {
+            LOGGER.error("Found invalid feature {} with value {}.", combined + key, featureValue.toString());
+        }
+
+        if (serverSideFeature != null && serverSideFeature.getId() == null) {
+            if (Map.class.isAssignableFrom(featureValue.getClass())) {
+                features = (Map<String, Object>) featureValue;
+                for (String fKey : features.keySet()) {
+                    addServerSideFeature(features, fKey, combined + key);
+                }
+            }
+        } else if (serverSideFeature != null) {
+            if (serverSideFeature.getConditions() != null && serverSideFeature.getConditions().getClientFilters() != null
+                && serverSideFeature.getConditions().getClientFilters().size() > 0) {
+                final Feature feature = new Feature();
+                feature.setKey(serverSideFeature.getId());
+                feature.setEvaluate(serverSideFeature.isEnabled());
+                feature.setEnabledFor(serverSideFeature.getConditions().getClientFilters());
+                feature.setRequirementType(serverSideFeature.getConditions().getRequirementType());
+                featureManagement.put(serverSideFeature.getId(), feature);
+            } else {
+                onOff.put(serverSideFeature.getId(), serverSideFeature.isEnabled());
+            }
+        }
+    }
+
+    // todo what is the best way to decide whether is backend schema?
+    private boolean isBackendSchema(Map<? extends String, ? extends Object> features) {
+        if (features.keySet().isEmpty()) {
```

Comment:
It's not a great option, but we could just check if the id field exists. It shouldn't exist for the other one. Also, maybe we default to this option, if we see it and we can fall back to the other option on error.

Reply:
That make sense. I updated the logic to try to parse the properties by server side schema at first. If the parse process failed or get no features after this parse process, then we need to try parse by client side schema.